### PR TITLE
Property test vec new approach

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: property_tests/arbitraries/candid/constructed/opt_arb/base.ts [hunk 1]
```diff
@@ -1,69 +1,129 @@
...
+} from '../../definition_arb/types';
+import { CandidValueArb, CandidValues } from '../../values';
+import { CandidType } from '../../candid_type';
 
 type SomeOrNone = 'Some' | 'None';
 
-export function OptArb(
-    candidTypeArb: fc.Arbitrary<CandidMeta<CandidType>>
-): fc.Arbitrary<CandidMeta<Opt>> {
+export function OptDefinitionArb(
```
Comment (bdemann) [inline property_tests/arbitraries/candid/constructed/opt_arb/base.ts:15]: Break all of these into their own files

File: property_tests/arbitraries/candid/constructed/blob_arb/definition.ts
Comment (dansteren) [inline property_tests/arbitraries/candid/constructed/blob_arb/definition.ts:1]: Rename this to have _arb in the name.

File: property_tests/arbitraries/candid/constructed/blob_arb/values.ts
Comment (dansteren) [inline property_tests/arbitraries/candid/constructed/blob_arb/values.ts:1]: add _arb to name.

File: property_tests/arbitraries/candid/constructed/variant_arb/definition_arbs.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        selector: (entry) => entry[0],
+        minLength: 1
+        // Although no minLength is technically required (according to the
+        // spec), the DFX CLI itself currently errors out trying to pass
+        // an empty object.
+    });
+}
+
+function generateImports(fields: Field[]): Set<string> {
+    const fieldImports = fields.flatMap((field) => [
```
Comment (dansteren) [inline property_tests/arbitraries/candid/constructed/variant_arb/definition_arbs.ts:58]: Consider destructuring to make this more readable what field[1] is. This is done in multiple places so do a pass to try to catch all these. I.e. go find all occurrences of this.

```suggestion
    const fieldImports = fields.flatMap(([name, definition]) => [
```

**Review by dansteren — CHANGES_REQUESTED** [review]

File: property_tests/arbitraries/candid/value_and_meta_arb.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        typeAliasDeclarations: string[];
+        imports: Set<string>;
+        valueLiteral: string;
+    };
+};
+
+/**
+ * An arbitrary representing all possible Candid types.
+ *
+ * **Note:** This currently only supports ints, nats, and null arbitraries
```
Comment (lastmjs) [inline property_tests/arbitraries/candid/value_and_meta_arb.ts:49]: Delete this part of the comment

File: property_tests/arbitraries/candid/primitive/bool.ts [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+import { CandidValues } from '../values';
 
-export const BoolArb = CandidMetaArb(fc.boolean(), 'bool', booleanToSrcLiteral);
+export const BoolArb = SimpleCandidValueAndMetaArb(
+    fc.boolean(),
+    'bool',
+    booleanToSrcLiteral
+);
+
+export const BoolDefinitionArb: fc.Arbitrary<BoolCandidDefinition> =
```
Comment (lastmjs) [inline property_tests/arbitraries/candid/primitive/bool.ts:15]: Remove all of these static type definitions in all files

File: property_tests/arbitraries/candid/primitive/bool.ts [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
-import { CandidMetaArb } from '../candid_arb';
+import { SimpleCandidValueAndMetaArb } from '../simple_type_arbs/value_and_meta_arb';
+import { SimpleCandidDefinitionArb } from '../simple_type_arbs/definition_arb';
+import { SimpleCandidValuesArb } from '../simple_type_arbs/values_arb';
 import { booleanToSrcLiteral } from '../to_src_literal/boolean';
+import { BoolCandidDefinition } from '../definition_arb/types';
+import { CandidValues } from '../values';
 
-export const BoolArb = CandidMetaArb(fc.boolean(), 'bool', booleanToSrcLiteral);
+export const BoolArb = SimpleCandidValueAndMetaArb(
```
Comment (lastmjs) [inline property_tests/arbitraries/candid/primitive/bool.ts:9]: SimpleCandidValueAndMetaArb should cease to exist and be replaced with ComplexCandidValueAndMetaArb...rename it to CandidValueAndMetaArbCreator or something

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Review by lastmjs — APPROVED** [review]

**Pull request merged**